BidAutomaton test failure

While testing #533, the CI stumbled upon a test failure which looks completely unrelated to the PR itself. Following is the stack trace:

```
=== RUN   TestMaintainBids
time="2020-05-18T07:10:13Z" level=warning msg="could not send bid tx" error="method not registered" process=BidAutomaton
coverage: 81.8% of statements
panic: test timed out after 10m0s

goroutine 8 [running]:
testing.(*M).startAlarm.func1()
	/home/travis/.gimme/versions/go1.13.11.linux.amd64/src/testing/testing.go:1377 +0x11c

created by time.goFunc

	/home/travis/.gimme/versions/go1.13.11.linux.amd64/src/time/sleep.go:168 +0x52

goroutine 1 [chan receive]:

testing.(*T).Run(0xc0000d4500, 0xfb2232, 0x10, 0xfd7578, 0x0)

	/home/travis/.gimme/versions/go1.13.11.linux.amd64/src/testing/testing.go:961 +0x68a

testing.runTests.func1(0xc0000d4500)

	/home/travis/.gimme/versions/go1.13.11.linux.amd64/src/testing/testing.go:1202 +0xa7

testing.tRunner(0xc0000d4500, 0xc000173cd8)

	/home/travis/.gimme/versions/go1.13.11.linux.amd64/src/testing/testing.go:909 +0x19a

testing.runTests(0xc00000fa80, 0x1697f90, 0x1, 0x1, 0x0)

	/home/travis/.gimme/versions/go1.13.11.linux.amd64/src/testing/testing.go:1200 +0x522

testing.(*M).Run(0xc000126280, 0x0)

	/home/travis/.gimme/versions/go1.13.11.linux.amd64/src/testing/testing.go:1117 +0x300

main.main()

	_testmain.go:94 +0x338

goroutine 7 [chan receive]:

github.com/dusk-network/dusk-blockchain/pkg/core/consensus/bidautomaton_test.TestMaintainBids(0xc0000d4600)

	/home/travis/gopath/src/github.com/dusk-network/dusk-blockchain/pkg/core/consensus/bidautomaton/bidautomaton_test.go:37 +0x2b6

testing.tRunner(0xc0000d4600, 0xfd7578)

	/home/travis/.gimme/versions/go1.13.11.linux.amd64/src/testing/testing.go:909 +0x19a

created by testing.(*T).Run

	/home/travis/.gimme/versions/go1.13.11.linux.amd64/src/testing/testing.go:960 +0x652

goroutine 18 [chan receive]:

github.com/dusk-network/dusk-blockchain/pkg/core/consensus/bidautomaton.(*BidAutomaton).Listen(0xc000194040)

	/home/travis/gopath/src/github.com/dusk-network/dusk-blockchain/pkg/core/consensus/bidautomaton/bidautomaton.go:65 +0xce

created by github.com/dusk-network/dusk-blockchain/pkg/core/consensus/bidautomaton.(*BidAutomaton).AutomateBids

	/home/travis/gopath/src/github.com/dusk-network/dusk-blockchain/pkg/core/consensus/bidautomaton/bidautomaton.go:57 +0x21b

goroutine 19 [chan receive]:

github.com/dusk-network/dusk-blockchain/pkg/core/consensus/bidautomaton_test.catchStakeRequest(0xc000190020, 0xc0001a2000)

	/home/travis/gopath/src/github.com/dusk-network/dusk-blockchain/pkg/core/consensus/bidautomaton/bidautomaton_test.go:79 +0x8b

created by github.com/dusk-network/dusk-blockchain/pkg/core/consensus/bidautomaton_test.TestMaintainBids

	/home/travis/gopath/src/github.com/dusk-network/dusk-blockchain/pkg/core/consensus/bidautomaton/bidautomaton_test.go:29 +0xf1

FAIL	github.com/dusk-network/dusk-blockchain/pkg/core/consensus/bidautomaton	600.024s
```

--- a/pkg/core/consensus/bidautomaton/bidautomaton_test.go
+++ b/pkg/core/consensus/bidautomaton/bidautomaton_test.go
@@ -26,7 +26,7 @@ func TestMaintainBids(t *testing.T) {
 	}()
 
 	c := make(chan struct{}, 1)
-	go catchStakeRequest(rb, c)
+	catchStakeRequest(rb, c)
 
 	// Send round update, to start the maintainer.
 	ru := consensus.MockRoundUpdate(1, nil)
@@ -41,7 +41,7 @@ func TestMaintainBids(t *testing.T) {
 	ruMsg = message.New(topics.RoundUpdate, ru)
 	bus.Publish(topics.RoundUpdate, ruMsg)
 
-	go catchStakeRequest(rb, c)
+	catchStakeRequest(rb, c)
 
 	select {
 	case <-c:
@@ -76,7 +76,9 @@ func catchStakeRequest(rb *rpcbus.RPCBus, respChan chan struct{}) {
 		panic(err)
 	}
 
-	<-c
-	respChan <- struct{}{}
-	rb.Deregister(topics.SendBidTx)
+	go func() {
+		<-c
+		respChan <- struct{}{}
+		rb.Deregister(topics.SendBidTx)
+	}()
 }

--- a/pkg/core/consensus/stakeautomaton/stakeautomaton_test.go
+++ b/pkg/core/consensus/stakeautomaton/stakeautomaton_test.go
@@ -26,7 +26,7 @@ func TestMaintainStakes(t *testing.T) {
 	}()
 
 	c := make(chan struct{}, 1)
-	go catchStakeRequest(rb, c)
+	catchStakeRequest(rb, c)
 
 	// Send round update, to start the maintainer.
 	ru := consensus.MockRoundUpdate(1, nil)
@@ -41,7 +41,7 @@ func TestMaintainStakes(t *testing.T) {
 	ruMsg = message.New(topics.RoundUpdate, ru)
 	bus.Publish(topics.RoundUpdate, ruMsg)
 
-	go catchStakeRequest(rb, c)
+	catchStakeRequest(rb, c)
 
 	select {
 	case <-c:
@@ -76,7 +76,9 @@ func catchStakeRequest(rb *rpcbus.RPCBus, respChan chan struct{}) {
 		panic(err)
 	}
 
-	<-c
-	respChan <- struct{}{}
-	rb.Deregister(topics.SendStakeTx)
+	go func() {
+		<-c
+		respChan <- struct{}{}
+		rb.Deregister(topics.SendStakeTx)
+	}()
 }

